### ISSUE TITLE
calling .wrapInline() twice causes breaking returns on both sides of selection

### ISSUE THREAD

**Issue by tyler-johnson** [ianstormtaylor/slate#185]
I have a simple transform like so:

```js
state.transform()
  .wrapInline("inner")
  .wrapInline("outer")
  .apply();
```

If before the transform I have a block that looks like this:

```
<div>
  Before Text After
</div>
```

I was expecting result of the transform to be this:

```
<div>
  Before
  <span class="outer">
    <span class="inner">Text</span>
  </span>
  After
</div>
```

However what I got looks like this:

```
<div>
  Before
  <span class="inner"><br /></span>
  <span class="outer">
    <span class="inner">Text</span>
  </span>
  <span class="inner"><br /></span>
  After
</div>
```

Am I doing something wrong here?

**Comment by ianstormtaylor** [comment]
Darn. Thanks for reporting this! 

It looks like there's two things going wrong:

1. The `<Leaf>` logic to add `<br />` elements in empty nodes to keep `contenteditable` from doing weird things needs to only apply to `Text` nodes that are direct children of `Block` nodes.

2. `wrapInline` is creating extra inline nodes from splitting the second time, since `splitInlineAtRange` doesn't actually check to see if it _needs_ to split before splitting, resulting in the empty `''` text being split off on either side.

First one should be easy to fix, I can get that going now. 

The second one is going to be a bit more mindbending to fix. The reason I cheated there in the first place was to avoid having to think through all of the cases of when the selection should and shouldn't be split. But it needs to happen!

**Comment by tyler-johnson** [comment]
Cool thanks! The first one is what I need to stop it from being forced onto its own line, so this works out.

**Comment by ianstormtaylor** [comment]
Fixed in `0.8.0`, but read the [`History.md`](https://github.com/ianstormtaylor/slate/blob/master/History.md) because there were a few changes. Most importantly:

1. `onKeyDown` and `onBeforeInput` have a different signature.
2. `Inline` nodes with `''` empty text are now prohibited and will be removed when `normalize` is run, so you'll never deal with them. I realized this was needed to match something similar to `contenteditable`'s native behavior. (It still has them, but deleting around them is very awkward, and they are not something you can rely on natively.)

So basically both problems solved, but solved in a different way. So it should work as expected now! Let me know if you run into anything still.
